spiffs_file_preprocessor/txt_processor.c

```diff
@@ -7,6 +7,20 @@
 #include "heatshrink_encoder.h"
 #include "fileUtils.h"
 
+long remove_chars(char* str, long len, char c) {
+    char *strReadPtr = str, *strWritePtr = str;
+    long newLen = 1;
+    for(long i = 0; i < len && *strReadPtr; i++)
+    {
+        *strWritePtr = *strReadPtr++;
+        strWritePtr += (*strWritePtr != c);
+        newLen += (*strWritePtr != c);
+    }
+    *strWritePtr = '\0';
+
+    return newLen;
```

Comment:
How did you test this? I spun up a trivial test program, and the length seems incorrect.
```C
#include <stdio.h>
#include <string.h>

long remove_chars(char* str, long len, char c) {
    char *strReadPtr = str, *strWritePtr = str;
    long newLen = 1;
    for(long i = 0; i < len && *strReadPtr; i++)
    {
        *strWritePtr = *strReadPtr++;
        strWritePtr += (*strWritePtr != c);
        newLen += (*strWritePtr != c);
    }
    *strWritePtr = '\0';

    return newLen;
}

int main()
{
    char rmv = 't';
    char tst[] = "test string";
    printf("remove '%c' from \"%s\"\n", rmv, tst);
    long newlen = remove_chars(tst, strlen(tst), rmv);
    printf("returned len: %d, strlen(): %d, str: \"%s\"\n", newlen, strlen(tst), tst);
    return 0;
}
```
prints
```
remove 't' from "test string"
returned len: 7, strlen(): 8, str: "es sring"
```

Reply:
It should be 9, so including the null terminator. There was a bug where a couple of lines were swapped, which caused chars to not be counted sometimes. Fixed and pushed.